# Basic Statistics Assignment
### 6031010021 Yanika Dontong

In [2]:
import statsmodels.api as sm
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

/usr/local/lib/python3.6/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


In [3]:
data = pd.read_csv('https://github.com/kaopanboonyuen/Python-Data-Science/raw/master/Dataset/adult.data.csv')
data.head()

age         workclass  fnlwgt  ... hours-per-week  native-country salary
0   39         State-gov   77516  ...             40   United-States  <=50K
1   50  Self-emp-not-inc   83311  ...             13   United-States  <=50K
2   38           Private  215646  ...             40   United-States  <=50K
3   53           Private  234721  ...             40   United-States  <=50K
4   28           Private  338409  ...             40            Cuba  <=50K

[5 rows x 15 columns]

![alt text](https://media.giphy.com/media/L20mbc7yRfsly/giphy.gif)

## Assignment 1: EDA

In [ ]:
# 1. How many men and women (sex feature) are represented in this dataset?

# You code here
woman = data[data['sex']=='Female']
man = data[data['sex']=='Male']
print("Woman: "+ str(len(woman)))
print("Man: "+ str(len(man)))

Woman: 10771
Man: 21790


In [ ]:
# 2. What is the average, min, max, std, and median age (age feature) of women?

# You code here
print(woman['age'].describe())


count    10771.000000
mean        36.858230
std         14.013697
min         17.000000
25%         25.000000
50%         35.000000
75%         46.000000
max         90.000000
Name: age, dtype: float64


**so, average is *36.858230*,
min is *17*,
max is *90*,
std is *14.013697*,
median is *35***

In [ ]:
# 3. What is the percentage of United-States citizens (native-country feature)?

# You code here
data_gb_country = data.groupby('native-country')['age'].count().reset_index()
data_gb_country['Percentage'] = 100 * data_gb_country['age']  / data_gb_country['age'].sum()
data_gb_country[data_gb_country['native-country']=='United-States']

native-country    age  Percentage
39  United-States  29170   89.585701

In [ ]:
# 4. What are the mean and standard deviation of age for those who earn more than 50K per year (salary feature) 

# You code here
data_gb_salary = data.groupby('salary')
data_gb_salary.get_group('>50K')['age'].describe()

count    7841.000000
mean       44.249841
std        10.519028
min        19.000000
25%        36.000000
50%        44.000000
75%        51.000000
max        90.000000
Name: age, dtype: float64

**so, average age is *44.249841*,
standard deviation is *10.519028***

In [ ]:
# 5. from 4 and those who earn less than 50K per year?

# You code here
data_gb_salary.get_group('<=50K')['age'].describe()

count    24720.000000
mean        36.783738
std         14.020088
min         17.000000
25%         25.000000
50%         34.000000
75%         46.000000
max         90.000000
Name: age, dtype: float64

**so, average age is *36.783738*,
standard deviation is *14.020088***

In [ ]:
# 6. Is it true that people who receive more than 50k have at least high school education? 
# (education - Bachelors, Prof-school, Assoc-acdm, Assoc-voc, Masters or Doctorate feature)

# You code here
data_salary_grt_50k_gb_edu = data[data['salary']==">50K"].groupby('education').count()
data_salary_grt_50k_gb_edu['age']

education
10th              62
11th              60
12th              33
1st-4th            6
5th-6th           16
7th-8th           40
9th               27
Assoc-acdm       265
Assoc-voc        361
Bachelors       2221
Doctorate        306
HS-grad         1675
Masters          959
Prof-school      423
Some-college    1387
Name: age, dtype: int64

It is **False**, as the result above people who has salary more than 50K have variant educations.

In [ ]:
# 7. What is the maximum number of hours a person works per week (hours-per-week feature)?

# You code here
max(data['hours-per-week'])

99

In [10]:
# 8. How many people work such a number of hours and what is the percentage of those who earn a lot among them?

# You code here
nine_hours_worker = data[data["hours-per-week"]==9]
rich_nine_hours_worker=nine_hours_worker[nine_hours_worker['salary']=='>50K']
rich_percentage = len(rich_nine_hours_worker)*100/len(nine_hours_worker)
print("there are "+str(len(nine_hours_worker)) + " workers who have worked 9 hours-per-week")
print("there are "+str("{:.2f}".format(rich_percentage))+" percent of them earn more salary")

there are 18 workers who have worked 9 hours-per-week
there are 5.56 percent of them earn more salary


## Assignment 2: Two sample t-test

Is there a significant difference between the population proportions of parents of black children and parents of Hispanic children who report that their child has had some swimming lessons?

**Populations**: All parents of black children age 6-18 and all parents of Hispanic children age 6-18  
**Parameter of Interest**: p1 - p2, where p1 = black and p2 = hispanic  

**Null Hypothesis:** p1 - p2 = 0  
**Alternative Hypthosis:** p1 - p2 $\neq$ = 0  

**Data**: 247 Parents of Black Children. 36.8% of parents report that their child has had some swimming lessons. 
<br>308 Parents of Hispanic Children. 38.9% of parents report that their child has had some swimming lessons.

> Use of `ttest_ind()` from `statsmodels`
Difference in population proportion needs t-test. Also, the population follow a binomial distribution here. We can just pass on the two population quantities with the appropriate binomial distribution parameters to the t-test function.

The function returns three values: (a) test statisic, (b) p-value of the t-test, and (c) degrees of freedom used in the t-test.

In [19]:
# You code here
from scipy.stats import ttest_ind,levene

population1 = np.random.binomial(1, 0.368, 247)
population2 = np.random.binomial(1, 0.389, 308)
df = pd.DataFrame(data=population1)
df.to_csv("population1.csv", sep=',',index=False,header=None)
df = pd.DataFrame(data=population2)
df.to_csv("population2.csv", sep=',',index=False,header=None)

In [20]:
p1 = np.genfromtxt("population1.csv",  delimiter=",")
p2 = np.genfromtxt("population2.csv",  delimiter=",")
print("p1 data :-\n")
print(p1)
print("\n")
print("p2 data :-\n")
print(p2)

p1 data :-

[0. 0. 1. 0. 0. 0. 0. 0. 1. 0. 0. 0. 0. 0. 1. 0. 0. 1. 0. 0. 1. 1. 0. 1.
 0. 0. 0. 0. 0. 0. 0. 1. 0. 0. 0. 0. 1. 1. 1. 0. 1. 0. 1. 1. 0. 0. 0. 0.
 0. 0. 1. 0. 0. 0. 0. 1. 0. 0. 0. 1. 0. 0. 0. 0. 1. 0. 1. 0. 0. 0. 0. 0.
 0. 1. 1. 0. 0. 0. 0. 1. 0. 1. 0. 0. 0. 1. 1. 0. 1. 0. 1. 0. 0. 0. 1. 1.
 0. 0. 0. 0. 1. 0. 1. 0. 0. 1. 1. 1. 0. 0. 0. 0. 1. 0. 0. 0. 0. 0. 1. 0.
 0. 1. 0. 1. 1. 0. 1. 1. 0. 0. 0. 0. 0. 0. 1. 1. 0. 0. 1. 1. 0. 1. 0. 1.
 1. 1. 1. 1. 0. 1. 0. 1. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 1. 1. 0. 0. 1.
 0. 0. 1. 0. 0. 0. 1. 0. 1. 0. 0. 1. 1. 1. 1. 0. 1. 0. 1. 0. 0. 1. 1. 1.
 1. 0. 0. 1. 1. 0. 0. 1. 1. 0. 1. 1. 0. 0. 1. 0. 0. 1. 0. 1. 0. 1. 1. 0.
 1. 0. 1. 0. 0. 0. 1. 0. 1. 0. 0. 1. 1. 0. 0. 1. 0. 0. 0. 1. 1. 0. 0. 0.
 1. 0. 1. 0. 1. 0. 1.]


p2 data :-

[1. 0. 1. 1. 0. 0. 0. 1. 0. 0. 1. 1. 1. 0. 0. 0. 0. 0. 1. 0. 0. 0. 0. 0.
 1. 1. 1. 0. 0. 0. 1. 0. 0. 0. 0. 0. 0. 1. 1. 0. 0. 0. 0. 1. 0. 0. 0. 0.
 0. 0. 1. 1. 1. 0. 1. 1. 0. 0. 0. 0. 1. 0. 0. 1. 0. 1. 0. 0. 1. 0. 0. 1.
 1

In [30]:
ttest,pvalue = ttest_ind(p1,p2,equal_var=True) 
print("ttest: "+str(ttest))
print("pvalue: "+str(pvalue))
dgf = p1.size+p2.size-2
print("degree of freedom: "+ str(dgf))

if pvalue < 0.05:    
   print("we are rejecting null hypothesis")
else:
  print("we are accepting null hypothesis")

ttest: -0.15886634692036355
pvalue: 0.8738321756028915
degree of freedom: 553
we are accepting null hypothesis


**So, There is a significant difference between the population proportions of parents of black children and parents of Hispanic children who report that their child has had some swimming lessons**

## Assignment 3: One sample t-test

Let's say a cartwheeling competition was organized for some adults. The data looks like following,

(80.57, 98.96, 85.28, 83.83, 69.94, 89.59, 91.09, 66.25, 91.21, 82.7 , 73.54, 81.99, 54.01, 82.89, 75.88, 98.32, 107.2 , 85.53, 79.08, 84.3 , 89.32, 86.35, 78.98, 92.26, 87.01)

Is distance Is the average cartwheel distance (in inches) for adults more than 80 inches?

**Population**: All adults  
**Parameter of Interest**: $\mu$, population mean cartwheel distance.

**Null Hypothesis:** $\mu$ = 80 
<br>**Alternative Hypthosis**: $\mu$ > 80

**Data**:
<br>25 adult participants. 
<br>$\mu = 83.84$
<br>$\sigma = 10.72$

In [ ]:
# You code here
from scipy.stats import ttest_1samp
dist_list = [80.57, 98.96, 85.28, 83.83, 69.94, 89.59, 91.09, 66.25, 91.21, 82.7 , 73.54, 81.99, 54.01, 82.89, 75.88, 98.32, 107.2 , 85.53, 79.08, 84.3 , 89.32, 86.35, 78.98, 92.26, 87.01]
df = pd.DataFrame(data=dist_list)
df.to_csv("distance.csv", sep=',',index=False,header=None)

In [ ]:
distances = np.genfromtxt('distance.csv')
distances_mean = np.mean(distances)
print(distances_mean)
t,pval = ttest_1samp(distances,80)
print("t = ",t,", p-value = ",pval)

if pval < 0.05:    
   print("we are rejecting null hypothesis")
else:
  print("we are accepting null hypothesis")

83.84320000000001
t =  1.756973189172546 , p-value =  0.09167485698904507
we are accepting null hypothesis


**So, the average distance of cartwheel  (in inches) for adults are more than 80 inches.**
